### DISCUSSION TITLE
An option to remove all files from target folder prior to updating

### DISCUSSION THREAD

**Discussion by rossbar** [JamesIves/github-pages-deploy-action#565] [Q&A]
Thanks for the nice action!

One thing we noticed when trying this out was that if there were [leftover tracked files from the old deploy system](https://github.com/networkx/documentation/tree/gh-pages/latest) (see the files with the different commit message). In the old deploy script, we did an `rm -rf latest/` prior to copying over the freshly built docs. Is there a way we could replicate this behavior with this action? I may be missing something from my read of the docs, but it doesn't seem like there's a straightforward way to do so.

OTOH maybe we just need to clean this up manually once then the `clean=True` option will keep this from being a problem moving forward?

**Comment by JamesIves** [comment]
You can try doing the manual option and seeing if clean works going forward, but if it doesn't I'm wondering if some additional processing needs to take place so it knows to clean up the target folder too. 

Let me know if it works, otherwise I'll create a bug for it and try to get it fixed for the version 4 release.
